src/Work/CompanyWork.tsx

```diff
@@ -53,8 +54,8 @@ export class CompanyWork extends Work {
     influenceStockThroughCompanyWork(company, gains.reputation, cycles);
     return false;
   }
-  finish(): void {
-    if (!this.singularity) {
+  finish(cancelled: boolean, suppressDialog?: boolean): void {
+    if (!this.singularity && !suppressDialog && Settings.SuppressAugWorkFinishedPopup) {
       dialogBoxCreate(
```

Comment:
Same as above, Settings.SupressAugWorkFinishedPopup is being used for the opposite condition (if it's true, it's allowing creation of the dialog box instead of suppressing)

Same comment the other work types as well. But given the suggestion to remove the option altogether, addressing this will just be removing the extra condition instead of fixing the logic of it.